pyproject.toml

```diff
@@ -1,3 +1,57 @@
 [build-system]
 requires = ["setuptools"]
 build-backend = "setuptools.build_meta"
+
+[project]
+name = "cubed"
+version = "0.6.0"
+authors = [
+    {name = "Tom White", email = "tom.e.white@gmail.com"},
+]
+license = {text = "Apache License 2.0"}
+description = "Bounded-memory serverless distributed N-dimensional array processing"
+readme = {file = "README.md", content-type = "text/markdown"}
+classifiers = [
+    "Development Status :: 3 - Alpha",
+    "License :: OSI Approved :: Apache Software License",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.8",
+]
+requires-python = ">=3.8"
+dependencies = [
+    "aiostream",
+    "fsspec",
+    "mypy_extensions", # for rechunker
+    "networkx < 2.8.3",
+    "numpy >= 1.22",
+    "pydot",
+    "tenacity",
+    "toolz",
+    "zarr",
+]
+
+[project.optional-dependencies]
+persist = ["dill"]
```

Comment:
I'm not sure we need a separate group for this - it's a pretty obscure use case (serializing the state of a computation). So just add `dill` to `test`?